test/gnmi/offlineintopotest.go

```diff
@@ -0,0 +1,114 @@
+	deviceClient, deviceClientError := env.Topo().NewDeviceServiceClient()
+	assert.NotNil(t, deviceClient)
+	assert.Nil(t, deviceClientError)
+	timeout := 10 * time.Second
+	newDevice := &device.Device{
+		ID:      offlineInTopoModDeviceName,
+		Address: offlineInTopoModDeviceName + ":11161",
+		Type:    offlineInTopoModDeviceType,
+		Version: offlineInTopoModDeviceVersion,
+		Timeout: &timeout,
+		TLS: device.TlsConfig{
+			Plain: true,
+		},
+	}
+	addRequest := &device.AddRequest{Device: newDevice}
+	addResponse, addResponseError := deviceClient.Add(context.Background(), addRequest)
+	assert.NotNil(t, addResponse)
+	assert.Nil(t, addResponseError)
+
+	// Make a GNMI client to use for requests
+	c, err := env.Config().NewGNMIClient()
+	assert.NoError(t, err)
+	assert.True(t, c != nil, "Fetching client returned nil")
+
+	// Set a value using gNMI client to the offline device
+
+	setPath := makeDevicePath(offlineInTopoModDeviceName, offlineInTopoModPath)
+	setPath[0].pathDataValue = offlineInTopoModValue
+	setPath[0].pathDataType = StringVal
+
+	_, extensionsSet, errorSet := gNMISet(MakeContext(), c, setPath, noPaths, noExtensions)
+	assert.NoError(t, errorSet)
+	assert.Equal(t, 1, len(extensionsSet))
+	extensionBefore := extensionsSet[0].GetRegisteredExt()
+	assert.Equal(t, extensionBefore.Id.String(), strconv.Itoa(gnmi.GnmiExtensionNetwkChangeID))
+	networkChangeID := network.ID(extensionBefore.Msg)
+
+	// Check that the value was set correctly
+	valueAfter, extensions, errorAfter := gNMIGet(MakeContext(), c, makeDevicePath(offlineInTopoModDeviceName, offlineInTopoModPath))
+	assert.NoError(t, errorAfter)
+	assert.Equal(t, 0, len(extensions))
+	assert.NotEqual(t, "", valueAfter, "Query after set returned an error: %s\n", errorAfter)
+	assert.Equal(t, offlineInTopoModValue, valueAfter[0].pathDataValue, "Query after set returned the wrong value: %s\n", valueAfter)
+
+	// Check for pending state on the network change
+	changeServiceClient, changeServiceClientErr := env.Config().NewChangeServiceClient()
+	assert.Nil(t, changeServiceClientErr)
+	assert.True(t, changeServiceClient != nil)
+	listNetworkChangeRequest := &diags.ListNetworkChangeRequest{
+		Subscribe:     true,
+		ChangeID:      networkChangeID,
+		WithoutReplay: false,
+	}
+	listNetworkChangesClient, listNetworkChangesClientErr := changeServiceClient.ListNetworkChanges(context.Background(), listNetworkChangeRequest)
+	assert.Nil(t, listNetworkChangesClientErr)
+	assert.True(t, listNetworkChangesClient != nil)
+	networkChangeResponse, networkChangeResponseErr := listNetworkChangesClient.Recv()
+	assert.Nil(t, networkChangeResponseErr)
+	assert.True(t, networkChangeResponse != nil)
+	assert.Equal(t, change.State_PENDING, networkChangeResponse.Change.Status.State)
+
+	// Start the device simulator
+	simulator := env.NewSimulator().SetName(offlineInTopoModDeviceName).SetAddDevice(false)
+	simulatorEnv := simulator.AddOrDie()
+	time.Sleep(100 * time.Second)
```

Comment:
This shouldn't be needed

Reply:
Nvm it is needed otherwise waiting for the network change times out. We'll just need a separate PR to address that and remove this sleep.